frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -1069,10 +1096,31 @@ export const funnelLogic = kea<funnelLogicType>({
             actions.loadEventWithPropertyCorrelations(eventName)
         },
 
-        excludeProperty: () => {
-            actions.setPropertyNames(
-                values.propertyNames.filter((property) => !values.excludedPropertyNames.includes(property))
-            )
+        excludePropertyFromProject: ({ propertyName }) => {
+            // When we exclude a property, we want to update the config stored
+            // on the current Team/Project.
+            const oldExcludedPropertyNames = values.excludedPropertyNames
+            const oldCurrentTeam = teamLogic.values.currentTeam
+
+            // If we haven't actually retrieved the current team, we can't
+            // update the config.
+            if (oldCurrentTeam === null || oldExcludedPropertyNames === null) {
+                console.warn('Attempt to update correlation config without first retrieving existing config')
+                return
+            }
+
+            const oldCorrelationConfig = oldCurrentTeam.correlation_config
+
+            const excludedPropertyNames = [...Array.from(new Set(oldExcludedPropertyNames.concat([propertyName])))]
```

Comment:
Can be a follow up PR: Thoughts on not actually firing the action when old propertyNames are same as new PropertyNames ?